src/Elmish/Hooks/Type.js

```diff
@@ -0,0 +1,11 @@
+const uuidV5 = require('uuid/v5')
+
+exports.genStableUUID = () => {
+  const stack = new Error().stack
+  const stackLines = stack.split('\n')
+  // TODO: Add tests to ensure this stays correct: See
+  // https://github.com/collegevine/purescript-elmish-hooks/pull/3#discussion_r774324284
+  // for an example.
+  const hookCallSite = stackLines[stackLines.length - 3]
```

Comment:
This is exactly what I had in mind, but I have to agree that this is wobbly. With respect to the number of frames to skip.

Here's a suggestion: make `mkHook` accept a `HookName` as before (or just a `ComponentName`. no need for a new type), and obligate individual hooks to call `genStableUUID`. Also make `genStableUUID` accept the number of skipped frames as a parameter. This way the individual hooks would have more control over how this works and more freedom in how to organize their own internal code.

Reply:
Ah, so then if you make a custom hook, you’d need to think about how many stack frames there are between `genStableUUID` and the call site of your hook. But in return, the builtin hooks are a lot easier to use. I wonder if there’s an API that makes it possible to guarantee a way to find the right stack frame.

But separately, I don’t know if I understand the `HookName` part. How does that help?

Reply:
> I don’t know if I understand the HookName part.

I don't understand what you don't understand :-)

Reply:
😂 I mean this part:

> make mkHook accept a HookName as before (or just a ComponentName. no need for a new type)

What does this help guarantee that `genStableUUID` doesn’t already?

Reply:
It's about pushing the responsibility for generating names to the hook. So that it's the hook that calls `genStableUUID`, not `mkHook`.

Reply:
Oh right, sorry, I misunderstood how the two were related. Got it. So if you wanted, you could write a custom hook that hardcodes a name or that makes consumers pass in a name. We’re not enforcing you use `genStableUUID`. That makes sense. One annoying thing is it would be difficult to figure out the number to pass to `genStableUUID` from a custom hook, but I guess that’s not a huge deal if you don’t _have_ to use `genStableUUID`.

Reply:
Yes, picking the right number would be tricky, but I think this is ok, because under normal circumstances you wouldn't create new hooks left and right, it would be a rare occasion, on the level of low-level library. 

Also, I think we can come up with some reasonable guidelines. For example: (1) always bind the unique name in a `where`, and (2) if your hook has more than one parameter, make it take one parameter anyway, and return a lambda. Under these circumstances, I think the number would always be 2 (one for the hook itself and one for whoever called the hook), e.g.:

```
useState initialState = mkHook name ...
  where
    name = genStableUUID { skipFrames: 2 }

fancyHook arg1 = \arg2 arg3 -> mkHook name ...
  where
    name = genStableUUID { skipFrames: 2 }
```

Reply:
One annoying point is that there is no way (that I can see) to detect the wrong number and fail early. You'd have to wait for weird bugs to appear.

Reply:
Maybe we can trace the stack frame that we picked, and thus the hook author could verify that it points to the right spot?

Reply:
Yeah that’s a good idea. Do you mean offer a separate function like `genStableUUIDWithTrace`?

Reply:
Yes, that. Or a parameter. I think a separate function is better.

Reply:
Cool, sounds good — thanks for the suggestions!